activesupport/lib/active_support/inflector/transliterate.rb

```diff
@@ -62,10 +62,12 @@ module Inflector
     # Transliteration is restricted to UTF-8, US-ASCII, and GB18030 strings.
     # Other encodings will raise an ArgumentError.
     def transliterate(string, replacement = "?", locale: nil)
-      string = string.dup if string.frozen?
       raise ArgumentError, "Can only transliterate strings. Received #{string.class.name}" unless string.is_a?(String)
       raise ArgumentError, "Cannot transliterate strings with #{string.encoding} encoding" unless ALLOWED_ENCODINGS_FOR_TRANSLITERATE.include?(string.encoding)
 
+      return string if string.ascii_only?
```

Comment:
This is technically a breaking change. It means `transliterate("foo".freeze)` now returns a frozen string while it would return a mutable one before.

Also `frozen_string.dup` is really cheap because it uses copy on write, so might not be worth changing this behavior.